### ISSUE TITLE
Help with basic usage and syntax errors

### ISSUE THREAD

**Issue by DrTron** [kellyjonbrazil/jello#51]
All,

I'm trying to something rather basic: extract values from lm_sensors, which I can export as json-formatted data.
However, I keep running into syntax errors I don't understand.

This is a small extract of the json-data:

```
{
  "aquaero-hid-3-1cb1": {
    "Adapter": "HID adapter",
    "Fan 1 voltage": {
      "in0_input": 4.24
    }
  }
}
```

jello -s gives me:
```
_ = {};
_.aquaero-hid-3-1cb4 = {};
_.aquaero-hid-3-1cb4.Adapter = "HID adapter";
_.aquaero-hid-3-1cb4["Fan 1 voltage"] = {};
_.aquaero-hid-3-1cb4["Fan 1 voltage"].in0_input = 4.24;
```

As expected. So to get the value for the Fan1 voltage, I try:
`cat x.json | jello _.aquaero-hid-3-1cb4["Fan 1 voltage"].in0_input` and
`cat x.json | jello '_.aquaero-hid-3-1cb4["Fan 1 voltage"].in0_input'`
That, however, gives:
```
jello:  Query Exception:  SyntaxError
        invalid syntax (<unknown>, line 1)
        SyntaxError:  _.aquaero-hid-3-1cb4[Fan 1 voltage].in0_input
        query:  _.aquaero-hid-3-1cb4[Fan 1 voltage].in0_input
        data:  {'aquaero-hid-3-1cb4': {'Adapter': 'HID adapter', 'Fan 1 voltage': {'in0_input': 4.24}}}
```
And now I'm confused. Isn't the result of "jello -s" the concatenation of levels I need to get my result?
The same thing happens for names without spaces in them, btw.
Even trying to get the first level with 
`cat x.json | jello _.aquaero-hid-3-1cb4`
results in a Syntax error.

Thanks!

Edit: After some trial and error, it seem that the "-3-1" is something the python syntax doesn't like. I can escape the spaces, but have not found a way to get around the "-3-1"....

**Comment by kellyjonbrazil** [comment]
Hi there - thanks for reporting this! I think you are running into shell escaping issues. Could you try your `jello` query with single quotes around it? That way the shell won't remove the double quotes in the bracket notation:
```
cat x.json | jello '_.aquaero-hid-3-1cb4["Fan 1 voltage"].in0_input'
```

**Comment by kellyjonbrazil** [comment]
hmm... that is strange, for some reason `jello` doesn't like the first key in dot notation:
```
X ~ % cat test.json | jello '_.aquaero-hid-3-1cb4'
jello:  Query Exception:  SyntaxError
        invalid decimal literal (<unknown>, line 1)
        SyntaxError:  _.aquaero-hid-3-1cb4
        query:  _.aquaero-hid-3-1cb4
        data:  {'aquaero-hid-3-1cb1': {'Adapter': 'HID adapter', 'Fan 1 voltage': {'in0_input':
            4.24}}}

X ~ % cat test.json | jello '_'                   
{
  "aquaero-hid-3-1cb1": {
    "Adapter": "HID adapter",
    "Fan 1 voltage": {
      "in0_input": 4.24
    }
  }
}
√ ~ % cat test.json | jello '_["aquaero-hid-3-1cb1"]'
{
  "Adapter": "HID adapter",
  "Fan 1 voltage": {
    "in0_input": 4.24
  }
}
```
It works in bracket notation, but I'm not sure why. I'll have to investigate.

**Comment by kellyjonbrazil** [comment]
Something about the name of the key makes the python interpreter think that could be a valid number?

When I change the dashes to underscores it works. I've never really tried using dashes in dictionary keys before so I haven't run into this. Since keys are typically strings it probably doesn't matter, but when using dot notation it's no longer a string but an object name in python so a dash in the name might not be legal.
```
√ ~ % cat test2.json| jello
{
  "aquaero_hid_3_1cb1": {
    "Adapter": "HID adapter",
    "Fan 1 voltage": {
      "in0_input": 4.24
    }
  }
}
√ ~ % cat test2.json| jello _.aquaero_hid_3_1cb1
{
  "Adapter": "HID adapter",
  "Fan 1 voltage": {
    "in0_input": 4.24
  }
}
```

**Comment by kellyjonbrazil** [comment]
I think I should fix the Schema output so it also encapsulates the key in brackets if it contains dashes. Looks like that is an easy fix.

**Comment by DrTron** [comment]
Thanks!
I was starting to think I was just too stupid to get simple commands working.
To be fair, "jq" also runs into an error. Probably also because issues with shell escaping.
But yes, the dashes seem to be what's causing the issue. Sadly, as the output of lm_sensors contains them, I have no way of getting rid of them.

**Comment by kellyjonbrazil** [comment]
No worries - can't believe I haven't run into this before. I will fix this and add tests. In the mean time, the best way forward would be to use bracket syntax for keys like that.

**Comment by kellyjonbrazil** [comment]
You can use bracket notation for all keys like this:
```
cat x.json | jello '_.["aquaero-hid-3-1cb4"]["Fan 1 voltage"]["in0_input"]'
```

**Comment by kellyjonbrazil** [comment]
I have a fix for this in the `dev` branch. The Schema output now checks for invalid or reserved keynames and converts them to bracket syntax.

**Comment by kellyjonbrazil** [comment]
fixed in v1.5.5.
